recipes/LibriSpeech/ASR/transformer/train_with_whisper.py

```diff
@@ -0,0 +1,357 @@
+
+        return loss
+
+    def fit_batch(self, batch):
+        should_step = self.step % self.grad_accumulation_factor == 0
+
+        # Managing automatic mixed precision
+        if self.auto_mix_prec:
+            self.whisper_optimizer.zero_grad()
+            with torch.cuda.amp.autocast():
+                outputs = self.compute_forward(batch, sb.Stage.TRAIN)
+            loss = self.compute_objectives(outputs, batch, sb.Stage.TRAIN)
+            self.scaler.scale(loss / self.grad_accumulation_factor).backward()
+            if should_step:
+                self.scaler.unscale_(self.whisper_optimizer)
+                if self.check_gradients(loss):
+                    self.scaler.step(self.whisper_optimizer)
+                self.scaler.update()
+                self.optimizer_step += 1
+        else:
+            outputs = self.compute_forward(batch, sb.Stage.TRAIN)
+            loss = self.compute_objectives(outputs, batch, sb.Stage.TRAIN)
+            loss.backward()
+            if self.check_gradients(loss):
+                self.whisper_optimizer.step()
+            self.whisper_optimizer.zero_grad()
+
+        return loss.detach()
+
+    def on_stage_start(self, stage, epoch):
+        """Gets called at the beginning of each epoch"""
+        if stage != sb.Stage.TRAIN:
+            self.cer_metric = self.hparams.cer_computer()
+            self.wer_metric = self.hparams.error_rate_computer()
+
+    def on_stage_end(self, stage, stage_loss, epoch):
+        """Gets called at the end of an epoch."""
+        # Compute/store important stats
+        stage_stats = {"loss": stage_loss}
+        if stage == sb.Stage.TRAIN:
+            self.train_stats = stage_stats
+        else:
+            stage_stats["CER"] = self.cer_metric.summarize("error_rate")
+            stage_stats["WER"] = self.wer_metric.summarize("error_rate")
+
+        # Perform end-of-iteration things, like annealing, logging, etc.
+        if stage == sb.Stage.VALID:
+
+            old_lr_whisper, new_lr_whisper = self.hparams.lr_annealing_whisper(
+                stage_stats["loss"]
+            )
+
+            sb.nnet.schedulers.update_learning_rate(
+                self.whisper_optimizer, new_lr_whisper
+            )
+            self.hparams.train_logger.log_stats(
+                stats_meta={"epoch": epoch, "lr_whisper": old_lr_whisper},
+                train_stats=self.train_stats,
+                valid_stats=stage_stats,
+            )
+            self.checkpointer.save_and_keep_only(
+                meta={"WER": stage_stats["WER"]}, min_keys=["WER"],
+            )
+        elif stage == sb.Stage.TEST:
+            self.hparams.train_logger.log_stats(
+                stats_meta={"Epoch loaded": self.hparams.epoch_counter.current},
+                test_stats=stage_stats,
+            )
+            with open(self.hparams.wer_file, "w") as w:
+                self.wer_metric.write_stats(w)
+
+    def init_optimizers(self):
```

Comment:
Same here, I am not sure that we need to override this function in this recipe (we only have one optimizer that can be passed as an argument to the Brain class when initialised)

Reply:
Done.